### ISSUE TITLE
bug: `parseRange()` returns unexpected result

### ISSUE THREAD

**Issue by javihernant** [denoland/deno_std#4344]
**Describe the bug**
`parseRange('>1.2')` produces the following `Range` object:
```
[ [ { operator: ">", major: 2, minor: 0, patch: 0 } ] ]
```

**Expected behavior**
It should instead produce (according to [node-semver](https://github.com/npm/node-semver?tab=readme-ov-file)):
```
[ [ { operator: ">=", major: 1, minor: 3, patch: 0 } ] ]
```
Otherwise, versions 1.x  wouldn't be included in the range.

I'd like to work on fixing this!

**Comment by kt3k** [comment]
> **Expected behavior** It should instead produce (according to [node-semver](https://github.com/npm/node-semver?tab=readme-ov-file)):
> 
> ```
> [ [ { operator: ">=", major: 1, minor: 3, patch: 0 } ] ]
> ```

I tested with the below code:

```js
import * as semver from "npm:semver";
const range = new semver.Range(">1.2.0");
console.log(range);
```

and it printed:

```
Range {
  options: {},
  loose: false,
  includePrerelease: false,
  raw: ">1.2.0",
  set: [
    [
      Comparator {
        options: {},
        loose: false,
        operator: ">",
        semver: SemVer {
          options: {},
          loose: false,
          includePrerelease: false,
          raw: "1.2.0",
          major: 1,
          minor: 2,
          patch: 0,
          prerelease: [],
          build: [],
          version: "1.2.0"
        },
        value: ">1.2.0"
      }
    ]
  ],
  range: ">1.2.0"
}
```

So I think the expectation here is:
```
[ [ { operator: ">=", major: 1, minor: 2, patch: 0 } ] ]
```

**Comment by kt3k** [comment]
fixed by #4350
